Arelle/Arelle
# Use GitHub API to check for and download updates

#### Reason for change
closes #466 

#### Description of change
Version numbers have been removed from download links on arelle.org and GitHub provides an actual API for detecting the latest release.

#### Steps to Test
* Create a `arelle/_version.py` module with the content `version = '2.1.3'` to test different version numbers (older, same, and newer than the latest release) with a local checkout to verify updates can be downloaded (on either macos or windows).
* Tests should cover the rest

**review**:
@Arelle/arelle
